hpke.go

```diff
@@ -135,10 +188,24 @@ type hpkeContext struct {
 	infoHash  []byte `tls:"head=none"`
 }
 
-func keySchedule(suite CipherSuite, mode HPKEMode, pkR KEMPublicKey, zz, enc, info, psk, pskID, pkIm []byte) (key, nonce []byte, err error) {
-	err = verifyMode(suite, mode, psk, pskID, pkIm)
+type ContextParameters struct {
+	Context []byte
```

Comment:
I'm slightly tempted to capture the intermediate products that go into the context string, but I'll grant that you could pull them out of this string.

Reply:
I'm a little split on whether to include the keyContext and nonceContext values.  There's a risk of people including null terminators, for example.  But I guess that's just one bit.  Maybe we should check to make sure the spec is clear.